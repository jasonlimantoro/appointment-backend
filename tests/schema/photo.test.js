import gql from 'graphql-tag';
import mockEntries from '../../fixtures/entries';
import { createTestClientAndServer } from '../utils';

describe('photo', () => {
  it('createPhoto: should work', async () => {
    const { mutate, photoAPI } = createTestClientAndServer();
    const photo = {
      id: 'some-id',
      key: 'some-s3-key/to/file',
      entryId: mockEntries[0].id,
      createdAt: 'some-date-now',
    };

    const spiedService = jest
      .spyOn(photoAPI, 'create')
      .mockResolvedValue(photo);
    const CREATE_PHOTO = gql`
      mutation CreatePhoto($input: CreatePhotoInput!) {
        createPhoto(input: $input) {
          id
          key
          createdAt
        }
      }
    `;
    const res = await mutate({
      mutation: CREATE_PHOTO,
      variables: {
        input: { key: 'some-s3-key/to/file', entryId: mockEntries[0].id },
      },
    });
    expect(res).toMatchSnapshot();
    expect(spiedService).toBeCalledWith({
      key: photo.key,
      entryId: photo.entryId,
    });
  });
});
