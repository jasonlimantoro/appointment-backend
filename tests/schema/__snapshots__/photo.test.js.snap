// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`photo createPhoto: should work 1`] = `
Object {
  "data": Object {
    "createPhoto": Object {
      "createdAt": "some-date-now",
      "id": "some-id",
      "key": "some-s3-key/to/file",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
